ert/data/record/_record.py

```diff
@@ -0,0 +1,224 @@
+import pathlib
+from abc import ABC
+from collections import deque
+from enum import Enum
+from typing import Dict, List, Mapping, MutableMapping, Optional, Tuple, Union, cast
+
+from beartype import beartype
+from beartype.roar import BeartypeException  # type: ignore
+from pydantic import PositiveInt
+from ert.serialization import get_serializer
+
+number = Union[int, float]
+numerical_record_data = Union[
+    List[number],
+    Dict[str, number],
+    Dict[int, number],
+]
+blob_record_data = bytes
+record_data = Union[numerical_record_data, blob_record_data]
+record_collection = Tuple["Record", ...]
+RecordIndex = Tuple[Union[int, str], ...]
```

Comment:
Should that not be `Union[Tuple[int, ...], Tuple[str, ...]]`?

Reply:
Yeah, but it's the same as in `main`:

    RecordIndex = Tuple[Union[StrictInt, StrictStr], ...]

I can fix https://github.com/equinor/ert/issues/2067 which is what you're describing.

Reply:
Yes, that issue seems to be the same. Not sure if changing this will make `mypy` complain, but it could be an easy change.